uber/models/admin.py

```diff
@@ -71,23 +71,36 @@ def admin_email():
             return None
 
     @staticmethod
-    def access_set(id=None, include_read_only=False):
+    def get_access_set(id=None, include_read_only=False):
         try:
             from uber.models import Session
             with Session() as session:
                 id = id or cherrypy.session.get('account_id')
-                access_groups = session.admin_account(id).access_groups
-                access_list = [list(group.access) for group in access_groups]
+                account = session.admin_account(id)
                 if include_read_only:
-                    access_list = access_list + [list(group.read_only_access) for group in access_groups]
-                return set([item for sublist in access_list for item in sublist])
+                    return account.read_or_write_access_set
+                return account.write_access_set
         except Exception:
             return set()
 
     @classproperty
     def _extra_apply_attrs(cls):
         return set(['access_groups_ids'])
 
+    @property
+    def write_access_set(self):
+        access_list = [list(group.access) for group in self.access_groups]
+        return set([item for sublist in access_list for item in sublist])
```

Comment:
It annoys me that there is no ``itertools.flatten`` function in the standard library.